src/adapter/z-stack/adapter/zStackAdapter.ts

```diff
@@ -308,68 +321,176 @@ class ZStackAdapter extends Adapter {
     }
 
     public async nodeDescriptor(networkAddress: number): Promise<NodeDescriptor> {
-        return await this.queue.execute<NodeDescriptor>(async () => {
-            this.checkInterpanLock();
-            try {
-                const result = await this.nodeDescriptorInternal(networkAddress);
-                return result;
-            } catch (error) {
-                logger.debug(`Node descriptor request for '${networkAddress}' failed (${error}), retry`, NS);
-                // Doing a route discovery after simple descriptor request fails makes it succeed sometimes.
```

Comment:
The discovery bit? I made a note of that.

I'm thinking maybe adding a function to `Adapter` for failed requests, something like this: `onFailedRequest(ieeeAddress, networkAddress, profileId, clusterId)`, which might also allow to make the other various retry logic more generic in the future (upstream in Controller/Device instead of Adapter).
Thoughts?
I guess we could also have a `discoverRoute` function in Adapter (which may not do anything if not needed). This is quite stack-specific, emberznet does this automatically for any frame via APS option for example.
I guess we could temporarily add it back in `sendZdo`, I expect this would more or less do the same?
```typescript
            if (clusterId === Zdo.ClusterId.NODE_DESCRIPTOR_REQUEST) {
                await this.discoverRoute(networkAddress);
            }

            await this.znp.requestZdo(clusterId, payload, waiter?.ID);
```

Reply:
Yes let's add it to `sendZdo` for now.